src/test/java/org/dcsa/api_validator/ovs/v2/PostTimestampsTest.java

```diff
@@ -0,0 +1,588 @@
+
+        Map<String, String> map = (Map<String, String>) jsonToMap(VALID_TIMESTAMP);
+        map.remove("eventLocation");
+        map.remove("vesselPosition");
+        map.remove("modeOfTransport");
+        map.remove("portCallServiceTypeCode");
+        map.put("facilitySMDGCode","aBCDaFGHaE");
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(400);
+
+        map.put("facilitySMDGCode", "aBC");
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(400);
+    }
+
+    @Test
+    public void testEventLocationQueryParam() {
+
+        Map<String, String> map = (Map<String, String>) jsonToMap(VALID_TIMESTAMP);
+        map.remove("facilitySMDGCode");
+        map.remove("vesselPosition");
+        map.remove("modeOfTransport");
+        map.remove("portCallServiceTypeCode");
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                header("testname","testEventLocationQueryParam").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(204);
+    }
+
+    // Test EventLocation, fails as format is JSON.
+    @Test
+    public void testEventLocationFalseFormat() {
+
+        Map<String, String> map = (Map<String, String>) jsonToMap(VALID_TIMESTAMP);
+        map.remove("facilitySMDGCode");
+        map.remove("vesselPosition");
+        map.remove("modeOfTransport");
+        map.remove("portCallServiceTypeCode");
+
+        map.put("eventLocation","aBCDaFGHaE"); // Wrong format
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(400);
+
+        map.put("eventLocation", "{}"); // Empty (fails Maybe - does not include mandatory parameters)
+
+        given().
+                auth().
+                oauth2(Configuration.accessToken).
+                contentType("application/json").
+                body(map).
+                post(Configuration.ROOT_URI + "/timestamps").
+                then().
+                assertThat().
+                statusCode(400);
+    }
+
+    @Test
+    public void testVesselPositionQueryParam() {
```

Comment:
I think this test name is wrong - as it is not testing query parameters. It is testing vesselPosition field in a timestamp